nautobot/core/ui/object_detail.py

```diff
@@ -895,6 +899,74 @@ def render_body_content(self, context):
         return result
 
 
+class _BaseTextPanel(Panel):
+    """A panel that renders simple text or markdown"""
+
+    RENDER_OPTIONS = namedtuple("RENDER_OPTIONS", ["plaintext", "json", "yaml", "markdown"])
+
+    def __init__(
+        self,
+        *,
+        render_as=RENDER_OPTIONS.markdown,
+        body_content_template_path="components/panel/body_content_text.html",
+        render_placeholder=True,
+        **kwargs,
+    ):
+        self.render_as = render_as
+        self.render_placeholder = render_placeholder
+        super().__init__(body_content_template_path=body_content_template_path, **kwargs)
+
+    def render_body_content(self, context):
+        text_content = self.get_text(context)
+        render_as = self.render_as
+
+        if not text_content and self.render_placeholder:
+            text_content = HTML_NONE
+            render_as = self.RENDER_OPTIONS.plaintext
```

Comment:
It was my first choice to be honest, but then I thought what if in future we would like to add some wrapper around html generated by this panel and in such case it would be better to return placeholder "within" the template. But I bet that we can keep it simple for now and eventually update it in future :)